### PR TITLE
JDK8 or JDK 11 enforcement added to JDK11 branch

### PR DESCRIPTION
**Pull request by AricMonary**
# Pull Request Description

This pull request closes strongbox/strongbox#1865 . Relates to #1775 .

# Acceptance Test

* [X] Building the code of the [`strongbox-parent`](https://github.com/strongbox/strongbox-parent) with `mvn clean install` still works.
* [X] Building the code of the [`strongbox`](https://github.com/strongbox/strongbox) project with `mvn clean install -Dintegration.tests` still works.
* [X] Running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [X] Building the code and running the `strongbox-distribution` from a `zip` or `tar.gz` still works.
* [X] The tests in the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

# Questions

* Does this pull request break backward compatibility? 
  * [ ] Yes
  * [X] No

* Does this pull request require other pull requests to be merged first? 
  * [ ] Yes, please see #...
  * [X] No

* Does this require an update of the documentation?
  * [ ] Yes, please see strongbox/strongbox-docs#{PR_NUMBER}
  * [X] No

# Code Review And Pre-Merge Checklist

* [X] My code follows the [coding convention](https://strongbox.github.io/developer-guide/coding-convention.html) of this project.
* [X] I have performed a self-review of my own code.
* [X] I have commented my code in hard-to-understand areas.
* [X] My changes generate no new warnings.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -393,6 +399,47 @@
...
+                        <configuration>
+                            <rules>
+                                <requireJavaVersion>
+                                     <version>[1.8.0-241,1.9),[11.3, 12)</version>
+                                    <message>Invalid JDK version, minimum required is 1.8.0_241 for JDK 8 or 11.3 for JDK 11.</message>
+                                </requireJavaVersion>
+                            </rules>
+                        </configuration>
+                    </execution>
+				</executions>
```
Comment (carlspring) [inline pom.xml:441]:
```suggestion
                    </executions>
```

**Review by sbespalov — CHANGES_REQUESTED** [review]
@AricMonary  thanks for PR!
could you please cleanup a commits here? it should be just your commits against latest `issues-1775`. Commonly this could be done with rabase.

**Comment by sbespalov** [comment]
@AricMonary sorry, this was closed by accident
anyway the `strongbox:issues-1775` already merged into `issues/1649` branch and most likely you need to open new PR, also please keep the commits history clean there.

**Comment by carlspring** [comment]
@AricMonary ,

As @sbespalov 's JDK 11 changes are now merged into the `issues/1649` branch, could you please create a new pull request against that branch apply your fixes there? Thanks and apologies for the inconvenience! :)

**Pull request closed without merging**